### PR TITLE
Auto-generated trace library (GAPIC only)

### PR DESCRIPTION
**Pull request by liyanhui1228**

### CODE REVIEW

**Comment by dhermes** [comment]
ISTM you want to move `trace/docs/` to `docs/trace/`?

Also, please make PRs from branches in your own fork of this repo (rather then making branches directly in this repo).

**Comment by liyanhui1228** [comment]
@dhermes Yeah the docs should live in docs/trace. Will move it later.

And sorry for forgetting to make PRs from my own fork... I remember you told me in last PR though. I spent quite a while separating the previous large PR to two PRs this afternoon because of some weird errors and then forgot that.

#3513 is based on the trace-autogen branch, which is the manual layer on top of the autogen layer. So I think the merging order should be merging the manual one to this autogen branch first then merge the autogen branch to master.

**Comment by duggelz** [comment]
The trace team is working on adding v2 of the API, timeline hazy, but v1 is in reasonably heavy use and will presumably remain so.  The two apis are quite similar, just some renaming and slightly restructuring.  It seems reasonable to me to have autogen code for v1 now, add v2 later, and have the manual layer work for both.  But @lukesneeringer will have to say for sure.

**Comment by duggelz** [comment]
Also, my git (lack of skill) is entirely to blame for any problems with these two pull requests, although @liyanhui1228 was much too kind to say so :)  We spent a while trying to resurrect the previous pull request from a deleted branch.

**Comment by jonparrott** [comment]
Please don't hesitate to reach out for git issues. I'm a bit of a git
surgeon.

**Comment by liyanhui1228** [comment]
@lukesneeringer Are there any plan for updating the autogen code to use the v2 API? I have #3513 ongoing based on this PR, so if there are anything needed to be changed in the autogen code please let me know.

**Comment by lukesneeringer** [comment]
Hi @liyanhui1228,
How did you generate this code? I want to make sure I get proto docstrings added in, is why I ask. It should just entail re-generating with the newest Artman.

Happy to do it and update this PR; I just need to know what you based it off of.

**Comment by lukesneeringer** [comment]
> @lukesneeringer Are there any plan for updating the autogen code to use the v2 API? I have #3513 ongoing based on this PR, so if there are anything needed to be changed in the autogen code please let me know.

This is based off of v1, yes? Most likely they can live side by side.

**Comment by liyanhui1228** [comment]
@lukesneeringer This is just the one you generated before, I wasn't able to reopen that closed PR so I open a new one for the autogen code. It would be great if you regenerate and update this PR, I just need to know if there is any change for calling the APIs as my manual code PR is based on this.

File: docs/trace/conf.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+source_suffix = '.rst'
+
+# The encoding of source files.
+#source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'gapic-google-cloud-trace-v1'
```
Comment (lukesneeringer) [inline docs/trace/conf.py]: This should probably be manually renamed to `google-cloud-trace`.

File: docs/trace/conf.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+# If true, show URL addresses after external links.
+#man_show_urls = False
+
+# -- Options for Texinfo output -------------------------------------------
+
+# Grouping the document tree into Texinfo files. List of tuples
+# (source start file, target name, title, author,
+#  dir menu entry, description, category)
+texinfo_documents = [
+    (master_doc, 'gapic-google-cloud-trace-v1',
```
Comment (lukesneeringer) [inline docs/trace/conf.py]: Same here, and everywhere else the string `'gapic-google-cloud-trace-v1'` appears.

File: trace/google/cloud/gapic/trace/v1/trace_service_client.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+        # The library may or may not be set, depending on what is
+        # calling this client. Newer client libraries set the library name
+        # and version.
+        if lib_name:
+            metrics_headers[lib_name] = lib_version
+
+        # Finally, track the GAPIC package version.
+        metrics_headers['gapic'] = pkg_resources.get_distribution(
+            'google-cloud-trace', ).version
```
Comment (lukesneeringer) [inline trace/google/cloud/gapic/trace/v1/trace_service_client.py]: This one is right. :-)

File: trace/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+googleapis-common-protos>=1.5.2, <2.0dev
```
Comment (lukesneeringer) [inline trace/requirements.txt]: This entire file can be eliminated.

Comment (liyanhui1228) [inline trace/requirements.txt]: Removed.

File: docs/trace/conf.py (hunk 2 above)
Comment (liyanhui1228) [inline docs/trace/conf.py]: Done.

File: docs/trace/conf.py (hunk 1 above)
Comment (liyanhui1228) [inline docs/trace/conf.py]: Done.

**Comment by dhermes** [comment]
@liyanhui1228 @lukesneeringer This broke the `docs` build

**Comment by liyanhui1228** [comment]
@dhermes I'm looking at it right now.

**Comment by dhermes** [comment]
Thanks!

**Pull request merged**